docs/docs/proposals/003-grafanaserviceaccount-crd.md

```diff
@@ -7,76 +7,96 @@ linkTitle: "GrafanaServiceAccount CRD"
 
-### Defining what Grafana Service Account belongs to what grafana.
+Currently you are only able to create these grafana service accounts using the grafana GUI or by using the HTTP-API after the grafana has already been deployed and is running.
+When not using persistent storage, this service account is removed on reconciliation so there is no way to declaratively manage service accounts as code, using the operator.
 
-When placing them inside the Grafana CRD this is not an issue.
+## Proposal
 
-### Defining Grafana Service Account to Grafana operator.
+To support this functionality, we propose the following changes to the Grafana operator.
 
-Today the Grafana Service Account is only held in memory if not using persistent, so when a pod is restarted or redeployed the grafana service account is removed. The grafana service accounts are also only possible to create using the Grafana GUI or with the HTTP-API and cannot be pre-defined before deploy or kept as IAC. But with the token kept as a k8s-secret it would be possible for other applications to use that token even when its rotated or recreated by the operator. RBAC rules are often more restrictive in the view scope so having the k8s-secret with the token in the same namespace as the grafana instance and other running applications would be beneficial.
+### Create a new resource `GrafanaServiceAccount`
 
-> Proposed updated CRD for Grafana
+This resource controls the reconciliation of service accounts. An example could look like this:
 
 ```.yaml
 apiVersion: grafana.integreatly.org/v1beta1
-kind: Grafana
+kind: GrafanaServiceAccount
 metadata:
   name: grafana-sa
-  namespace: grafana-namespace
 spec:
-  grafanaServiceAccounts: #Not sure if this is the right place to place it but thats easily fixed when implementing.
-    createServiceAccount:
-        generateTokenSecret: [true/false] #Will create the k8s-secret with a default name if true. Defaults to true.
-    accounts: #Since its possible today to have multiple service accounts it should be a list of accounts.
-        - id: grafana-sa
-          name: grafana-service-account
-          roles: [Viewer/Editor/Admin]
-          tokens: #This is a list of the tokens that belongs to this GSA and that the operator should create k8s-secrets with tokens for with the names specified. If not specified it would default to creating a token in a k8s-secret with a default name if spec.createServiceAccount.generateTokenSecret is true.
-              - Name: grafana-sa-token-<name-of-GSA>
-              expires: <Absolute date for expiration, defaults to Never>
-          permissions:    #This is to try and match what values can be set when creating GSA in the GUI where you can set different permissions for users and groups.
-              - user: <users in the cluster/root user etc>
-              permission: [Edit/Admin]
+  instanceName: test-grafana-instance
+  name: grafana-service-account
+  role: Viewer # Valid options: Viewer, Editor, Admin
```

Comment:
Are there any more roles in Grafana Enterprise that we should take into consideration?